guava/src/com/google/common/primitives/UnsignedBytes.java

```diff
@@ -135,7 +136,7 @@ public static int compare(byte a, byte b) {
    *     the array
    * @throws IllegalArgumentException if {@code array} is empty
    */
-  public static byte min(byte @MinLen(1)... array) {
+  public static @Unsigned byte min(byte @Unsigned @MinLen(1)... array) {
```

Comment:
I was surprised to see the `@Unsigned` annotations, since the Javadoc does not specify that as the type.
However, the code does treat the argument as unsigned.

This is a nice find!

This seems like a deficiency in the documentation.  It's not a huge deal since this is in a class `UnsignedBytes`, but it is still inconsistent with the rest of the file and a reader could get confused when looking at the Javadoc in isolation.

I think it would be a good idea to submit a pull request to Guava to fix it.  Please pass that pull request past me first.  It should contain either an example of the behavior, or a test case, or a reference to a test case.

There may be other pull requests to submit as well, to fix problems in Guava.  (`max` just below is an obvious example.)  Please keep an eye out for them.

Important: Please keep a record of all problems that the Signedness Checker reveals (pull request URLs are good enough), because we will want to list them when we write up the successes of the Signedness Checker.

Reply:
@mernst 
I have issued a pull request in google/guava related to the impreciseness in the documentation.
Please review it-[link](https://github.com/google/guava/pull/3507)